File: 0300-longest-increasing-subsequence/0300-longest-increasing-subsequence.py
class Solution:
    def lengthOfLIS(self, nums: List[int]) -> int:

        dp = [1] * len(nums)

        for i in range(1,len(nums)):
            
            maxSub = float("-inf")
            for j in range(i):
                if nums[j] < nums[i]:
                    maxSub = max(maxSub, 1 + dp[j])
            
            dp[i] = max(dp[i], maxSub)
        
        return max(dp)



        

File: 0300-longest-increasing-subsequence/0300-longest-increasing-subsequence.cpp
class Solution {
public:
    int lengthOfLIS(vector<int>& nums) {
        int n = nums.size();
        if (nums.size() == 1) return 1;
        vector<int> dp(n, 1);
        int res = 0;
        for (int i = 1; i < n; i++) {
            for (int j = i - 1; j >= 0; j--) {
                if (nums[j] < nums[i]) {
                    dp[i] = max(dp[i], 1 + dp[j]);
                }
            }
            res = max(res, dp[i]);
        }
        return res;
    }
};